server/src/connectionHandlers/onCompletion.ts

```diff
@@ -462,6 +468,34 @@ function convertExtraSymbolsToCompletionItems (uri: string): CompletionItem[] {
   return completionItems
 }
 
+export const getRangeOfTextToReplace = (document: TextDocument, position: Position): Range => {
```

Comment:
You could specify what range it is to replace. I think it's a word right? So rename getWordRange. You don't even need to specify it's for replacement. This function doesn't know what the word range will be used for. That way we are more susceptible to reuse it in other contexts

Reply:
There's different definitions of what is a word. In this `onCompletion.ts` file, we already have `analyzer.wordAtPointFromTextPosition`, which would return the whole strings with its quotes.
For example, if the user has their cursor anywhere in the string on `LICENSE = "GPL-2.0-or-later & LGPL-2.1-or-later"`, it will return `"GPL-2.0-or-later & LGPL-2.1-or-later"`

Javascript regexes have an other definition of word boundary: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Word_boundary_assertion.
For example, in `"GPL-2.0-or-later & LGPL-2.1-or-later"`, the words are `GPL`, `2`, `0`, `or`, `later` and so on. It seems to be what happen by default, and this is not what we want. If you remove this function (with its TextEdit) and use completion while typing the word `later`, you'll end up with something like `GPL-2.0-or-GPL-2.0-or-later`.

For completion, I determined we want replace whatever is located between quotes and whitespaces around the cursor. Since I am not aware of any terminology for that sort of text, I thought it would make sense to call it "textToReplace".

Reply:
Word isn't appropriate since these expression have several of them. Those expressions are delimited by quotes and whitespace. Their specificity is that they cannot contain whitespaces. Some ideas:
 - text-span-without-whitespace
 - expression-no-space

Reply:
After "solving" many issues, I realized my version of neovim was not supporting language servers. I updated it and the BitBake language server seems to be launched now. However, I still fail to have completion or any of the extension's functionalities, and I see `"server_request: no handler found for"	"bitbake/getRecipeLocalFiles"` in the logs.

I tried to trigger completion with `ctrl+x`, `ctrl+p`, and many other random combinations.

My init.lua currently looks like this (I've been trying many things):
```lua
local vim = vim
local Plug = vim.fn['plug#']

vim.call('plug#begin')
Plug('neoclide/coc.nvim', {['branch'] = 'release'})
Plug('neovim/nvim-lspconfig')
vim.call('plug#end')

--vim.lsp.omnifunc()
vim.lsp.buf.hover()

vim.api.nvim_create_autocmd({ "BufEnter" }, {
  pattern = { "*.bb"},
  callback = function()
      print("BufEnter event triggered for bitbake file")
      vim.notify("Starting bitbake LSP", vim.log.levels.INFO)

      vim.lsp.start({
      name = "bitbake",
      cmd = { "language-server-bitbake", "--stdio" }
    })
  end,
})
```

`:checkhealth lsp` shows the following:
```
vim.lsp: require("vim.lsp.health").check()

- LSP log level : WARN
- Log path: /home/io/.local/state/nvim/lsp.log
- Log size: 38 KB

vim.lsp: Active Clients ~
- bitbake (id=1, root_dir=~/Downloads/nvim/v:null, attached_to=[1])

vim.lsp: File watcher ~
- file watching "(workspace/didChangeWatchedFiles)" disabled on all clients
```

Reply:
```
"server_request: no handler found for"	"bitbake/getRecipeLocalFiles"
```
Means the server is sending a request to the client, which isn't there so this throws. There's a completion for local files in SRC_URI which must be what's throwing here. It probably cancels the request because we don't catch this early enough. However it looks like we do in server.ts:209. In that case it probably creates an error log, but we properly catch it. It must be something else.

Reply:
SO in our case onCompletion.ts:104, recipeLocalFiles is expected to be undefined. I don't think this should impact LICENSEs? We should remove this condition, and make required changes if any.
```typescript
    const isVariableAllowedForLicenseCompletion = analyzer.isStringContentOfVariableAssignment(documentUri, wordPosition.line, wordPosition.character, variablesAllowedForLicenseCompletion)
    if (isVariableAllowedForLicenseCompletion && recipeLocalFiles !== undefined && word !== null) {
```

Reply:
Your setup looks to properly start the bitbake language server now since it's dsiplaying a log. Currently, the only completion that works is new variable assignments on empty lines. Don't you have that?

Reply:
oh. No, I don't even have that :/